server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -344,6 +345,18 @@ private void tryAllHandlers(final RestRequest request, final RestChannel channel
                         BytesRestResponse.
                             createSimpleErrorResponse(channel, BAD_REQUEST, "multiple values for single-valued header [" + name + "]."));
                     return;
+                } else if (name.equals(Task.TRACE_PARENT)) {
+                    String traceparent = distinctHeaderValues.get(0);
+                    // TODO rely on high performance `TraceContext` from apm-agent-java
+                    // This does little validation of the actual id values, since we only use them to persist in logging for now this
+                    // should be fine. Once we actually start instrumenting through apm-agent-java it will take care of validation.
+                    // Java has a fast path for single character regexes so this should not be as bad as its signature implies
+                    String[] tokens  = traceparent.split("-");
+                    if (tokens.length == 4) {
```

Comment:
Just as a suggestion: you could validate the length of the header. If it has the expected minimum length, use `java.lang.String#substring(int, int)` with the known offsets for the trace id and parent id.